File: 番外篇/disc/第2章/server/go/controllers/histories.go
// Go：controllers\histories.go
package controllers

import (
	"github.com/kataras/iris/v12"
	"xorm.io/xorm"
	"local/models"
	"fmt"
)

// 基地址：/histories
type HistoryController struct {
	Ctx iris.Context // HTTP请求上下文对象
	Db  *xorm.Engine
}

// GET：/histories
func (c *HistoryController) Get() {
	account := c.Ctx.Values().Get("user").(models.Account)
	fmt.Println(account)
	histories := make([]models.History, 0)
	c.Db.Desc("id").Limit(10).Find(&histories)
	c.Ctx.View("histories/index", map[string]interface{}{
		"title":     "历史记录",
		"path":      "/histories",
		"histories": histories,
		"role":      account.Role})
}

File: 番外篇/disc/第2章/server/node/controllers/histories.js
// JS：controllers\histories.js
"use strict"
import model from "../models/index.js"

export default {
  "GET ": async (ctx, next) => {
    const title = "历史记录"
      , db = await model.connect()
      , histories = await model.execute(db, "select * from history order by date_created desc limit 10")
    ctx.render("histories/index", { title, path: ctx.request.path, histories })
  }
}